lib/brainstem/query_strategies/filter_and_search.rb

```diff
@@ -2,24 +2,34 @@ module Brainstem
   module QueryStrategies
     class FilterAndSearch < BaseStrategy
       def execute(scope)
-        ordered_search_ids = run_search(scope, filter_includes.map(&:name))
-        scope = scope.where(id: ordered_search_ids)
+        scope, ordered_search_ids = run_search(scope, filter_includes.map(&:name))
         scope = @options[:primary_presenter].apply_filters_to_scope(scope, @options[:params], @options)
-        scope = @options[:primary_presenter].apply_ordering_to_scope(scope, @options[:params])
-        count = scope.count
-        scope = paginate(scope)
-        primary_models = evaluate_scope(scope)
+
+        if ordering?
+          count = scope.count
+          scope = paginate(scope)
+          scope = @options[:primary_presenter].apply_ordering_to_scope(scope, @options[:params])
```

Comment:
When we are ordering, shouldn't we apply ordering and then paginate?

Reply:
Oooo. Yes.

Reply:
Yeah, but `paginate` in this case only adds AREL things to the SQL query and it doesn't actually cut out the pages we want in that line.

Reply:
Nice, sounds good.